mixnet/node.py

```diff
@@ -79,96 +76,59 @@ def __init__(
         inbound_socket: PacketQueue,
         outbound_socket: PacketPayloadQueue,
     ):
-        super().__init__()
         self.encryption_private_key = encryption_private_key
         self.delay_rate_per_min = delay_rate_per_min
         self.inbound_socket = inbound_socket
         self.outbound_socket = outbound_socket
-        self.num_processing = AtomicInt(0)
+        self.packet_processing_tasks = set()
```

Comment:
What is this used for?

Reply:
This is just for keeping track of the number of tasks being processed, so that, in the test, we can check how many packets are staying in a mix node: https://github.com/logos-co/nomos-specs/pull/53/files#diff-b773b3f0f891e8bb9864e3aa1dbbef792c2f6cfcd0a7d9e4fce70f7ea6745598R127. In real implementations, we actually don't need this. I'll leave a comment in the code.

Reply:
I think this then should be part of just the testing code. It is a bit confusing if we mix specification with testing.

Reply:
Done c7b32fe. Now we don't have the `set` as a class field. Also, all calculations for testing are done within tests.